fix(curriculum): add a test to check for "character" variable declaration (#58877)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/660ef19b95d3308e7dd31bb6.md

# --hints--

The `character` variable declaration should not be changed.

```js
assert.isDefined(character);

> assert.isNotEmpty(character);

Reviewer: please check that the value was not changed instead of just checking that it is not empty

# --hints--

> The `character` variable declaration should not be changed.

Reviewer: ```suggestion
The `character` variable initialization should not be changed.
```

a little nit on terms and then we are good to go

> You should declare a variable `character` and assign it a value.

Reviewer: ```suggestion
The `character` variable declaration should not be changed.
```

```js

> assert.match(__helpers.removeJSComments(code), /let\s+character\s*=\s*[^;]+;/);

Reviewer: please check if the variable exists and that it has the right value, do not use regex when there are other methods
